perftest/simulations/src/gatling/scala/org/projectnessie/perftest/gatling/KeyListSpillingSimulation.scala

```diff
@@ -0,0 +1,244 @@
+   */
+  private val paddingBuilder: mutable.StringBuilder = new mutable.StringBuilder()
+  for (i <- 1 to 32) {
+    paddingBuilder.append("abcdefg_")
+  }
+  val padding: String = paddingBuilder.toString();
+
+  /**
+   * Configuration specific to this simulation and scenario.
+   */
+  private val params: KeyListSpillingParams = KeyListSpillingParams.fromSystemProperties()
+
+  /**
+   * Session key for the zero-indexed commit number associated with the current simulated Gatling user.
+   *
+   * Gatling initializes and increments this for us via [[repeat]].
+   */
+  private val commitIndexKey = "commitIndex"
+
+  /**
+   * Commit once to a branch conveyed via the [[Session]].
+   *
+   * Although the returned chain fragment will commit only once, the number of put operations in that commit is
+   * determined by [[KeyListSpillingParams.putsPerTestCommit]].
+   */
+  private def commitToBranch: ChainBuilder = {
+      exec(
+        nessie("TestBranchCommit")
+          .execute { (client, session) =>
+
+            // This is initialized and incremented on our behalf by Gatling (see buildScenario)
+            val commitIndex = session(commitIndexKey).as[Int]
+            val commitNum = commitIndex + 1
+            val testBranch = session("branch").as[Branch]
+            val userId = session.userId
+            val putOps: List[Operation] = List.range(0, params.putsPerTestCommit).map(
+              i => {
+                val key = ContentKey.of(s"${userId} ${commitIndex} ${i}" + padding)
+                val table = IcebergTable.of(s"metadata_${userId}_${commitNum}_${i}", 42, 43, 44, 45)
+                Put.of(key, table, null)
+              }
+            )
+
+            // Call the Nessie client operation to perform a commit with one or more put ops
+            val updatedBranch = client
+              .commitMultipleOperations()
+              .branch(testBranch)
+              .commitMeta(
+                CommitMeta.fromMessage(s"worker commit userId=${userId} testCommitNum=${commitNum}")
+              )
+              .operations(putOps.asJava)
+              .commit()
+
+            session.set("branch", updatedBranch)
+          }
+      )
+  }
+
+  /**
+   * Delete a test [[Branch]] (if it exists), then (re)create it.
+   *
+   * The test branch's name ends in the Gatling userId (as conveyed through the [[Session]].
+   */
+  private def dropAndCreateTestBranch: ChainBuilder = {
+    exec(
+      nessie(s"DeleteTestBranchIfExists")
+        .execute { (client, session) =>
+          val testBranchName = params.getTestBranchName(session)
+
+          // Get the test branch (we must know its hash to delete it)
+          try {
+            val testBranch: Reference = client
+              .getReference
+              .refName(testBranchName)
+              .get()
+              .asInstanceOf[Branch]
+
+            // Delete the test branch
+            if (null != testBranch) {
```

Comment:
The null-check's unnecessary. the get above either throws a NRFE or returns a reference.

Reply:
Applied